spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -109,10 +109,12 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       "spark.comet.cast.stringToTimestamp is disabled")
   }
 
-  ignore("cast string to timestamp") {
-    withSQLConf((CometConf.COMET_CAST_STRING_TO_TIMESTAMP.key, "true")) {
-      val values = Seq("2020-01-01T12:34:56.123456", "T2") ++ generateStrings(timestampPattern, 8)
-      castTest(values.toDF("a"), DataTypes.TimestampType)
+  test("cast string to timestamp") {
+    withSQLConf(
+      (SQLConf.SESSION_LOCAL_TIMEZONE.key -> "UTC"),
+      (CometConf.COMET_CAST_STRING_TO_TIMESTAMP.key -> "true")) {
+      val values = Seq("2020-01-01T12:34:56.123456", "T2", "-9?")
```

Comment:
Could you add all the working example formats to this test? The Rust tests are great but we need them here as well so that we test against all supported Spark versions for compatibility.

Reply:
added all the examples.

```
val values = Seq(
        "2020",
        "2020-01",
        "2020-01-01",
        "2020-01-01T12",
        "2020-01-01T12:34",
        "2020-01-01T12:34:56",
        "2020-01-01T12:34:56.123456",
        "T2",
        "-9?")
```